docs/40_smart-contract-guides/92_read-only-queries.md

```diff
@@ -0,0 +1,337 @@
+## Before you begin
+This tutorial requires the following:
+
+* The EOSIO platform software, version 2.2. or greater, click on this link for instructions on [installing EOSIO binaries](https://developers.eos.io/manuals/eos/latest/install/install-prebuilt-binaries)
+* You understand how to build and deploy smart contracts
+
+## Run a Local Single-Node Blockchain
+Run a local nodeos instance which produces blocks, creating a local single-node testnet, using these [instructions](https://developers.eos.io/manuals/eos/latest/nodeos/usage/development-environment/local-single-node-testnet).
+ 
+
+## Build a Smart Contract with a Read-Only Query 
+
+Create a simple smart contract which contains three read-only actions: 
+- `get` - Will return a value 
+- `getComp` - Will return a complex value
+- `getMore` - Will return a vector of values
+
+Define the actions as read-only with the `eosio::read_only` attribute. Click on the follwing link for more information [ABI code generator attributes explained](https://developers.eos.io/manuals/eosio.cdt/latest/best-practices/abi/abi-code-generator-attributes-explained)   
+
+
+```cpp
+#include <eosio/eosio.hpp>
+class [[eosio::contract]] hello : public eosio::contract {
+  public:
+
+     struct retVal {
+         std::string name;
+         uint32_t gender;
+         uint32_t age;
+      };
+ 
+      using eosio::contract::contract;
+
+
+      [[eosio::action, eosio::read_only]] std::string get() {
+         return "Bob";
+      }
+
+
+      [[eosio::action, eosio::read_only]] retVal getComp() {
+         retVal rv;
+
+         rv.name = "Bob";
+         rv.gender = 1;
+         rv.age = 99;
+
+         return rv;
+      }
+
+
+      [[eosio::action, eosio::read_only]] std::vector<retVal> getMore() {         
+         std::vector<retVal> ret;
+         
+         retVal rv;
+         
+         rv.name = "Bob";
+         rv.gender = 1;
+         rv.age = 99;
+         ret.push_back(rv);
+
+         rv.name = "Alice";
+         rv.gender = 0;
+         rv.age = 19;
+         ret.push_back(rv);
+
+         return ret;
+      }
+};
+```
+
+Build the smart contract and deploy the smart contract to a blockchain using [how to compile a smart contract via the CLI](https://developers.eos.io/manuals/eosio.cdt/latest/how-to-guides/compile/compile-a-contract-via-cli) and [how to deploy a smart contract.](https://developers.eos.io/manuals/eos/latest/cleos/how-to-guides/how-to-deploy-a-smart-contract) 
+
+## Call the Read-Only Query 
+Once you have deployed the smart contract you can test it by calling the actions and seeing the data returned. Assuming the smart contract is deployed to an account called _scholder_ and that you have an account named _bob_ to authorise calling the actions. 
+
+* Call get():
+
+```shell
+cleos push transaction --read-only -j '{"actions":[{"account":"scholder","name":"get","authorization":[{"actor":"bob","permission":"active"}],"data":""}]}' -p bob@active
+```
+
+The output should be similar to:
```

Comment:
'should' vs 'will' again.

Reply:
imperative is deemed incorrect/rude and I have been pulled up for using will before.

Reply:
My biggest problem with this and the following two is we're waffling twice.  Saying "similar to" is already eliminating the sense of a definite statement.  Adding "should" on top of that is redundant to the point of absurdity.

Reply:
On general principles, the imperative is not rude and is precisely correct when documenting software.  In software, the results are not open to discussion or negotiation.  You get what you get.  Using 'will' is completely appropriate.  In this case, you _will_ get a JSON structure that looks exactly like this one, every time you successfully execute your transaction.  It is not up for debate or wishful thinking or any other common human malady.  The choice of the phrase "similar to" was only to avoid enumerating the known differences, which will be things like different block numbers and block IDs.

Reply:
using `should` I do not find it absurd at all; we can't be 100% certain of the results the user will get cause maybe he missed one tiny thing and then he might get something different (not just in the blocks numbers and/or block IDs).

as for `in software the results are not open for discussion or negotiation` a beg to differ again. this is wishful thinking on one hand and on the other hand here we deal with a user that learns and that might make some mistakes and not get this outcome that we stipulate here. therefor I do not find it absurd at all to use this language: `The output should be similar to`.

I am not a native speaker that's why I am calling in @sunshineveg; can you please chime in here with your thoughts? thanks.

Reply:
I found more info related to this subject, which at least for me changes the perspective; on the link below, search for `should, should be`:
https://developers.google.com/style/word-list

so I guess now a better way would be 
`The output should be similar to` -> `The output will be similar to`